File: 290-word-pattern/290-word-pattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        String arr[] = s.split(" ");
        if(arr.length != pattern.length()){
            return false;
        }
        
        HashMap<String,Character> firstMap = new HashMap<> ();        
        HashMap<Character,Boolean> secondMap = new HashMap<> ();
        
for(int i =0;i<pattern.length();i++){
    char ch = pattern.charAt(i);
    String str = arr[i];
    if(firstMap.containsKey(str)){
            if(firstMap.get(str)!=ch){
                return false;
            }
        }
    else{
            if(secondMap.containsKey(ch)){
                return false;
            }
            else{
            firstMap.put(str,ch);
            secondMap.put(ch,true);
            }
     }
    }
    return true;
    }
}

File: 290-word-pattern/290-word-pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
  const stringAsArray = s.split(" ");
  if(pattern.length !== stringAsArray.length) return false;
  const stringToPatternMap = {};
  const patternToStringMap = {};
  
  for(let i = 0 ; i < pattern.length ; i++){
    const patternChar = pattern[i];
    const str = stringAsArray[i];
    
    if(!(str in stringToPatternMap) && patternChar in patternToStringMap)
      return false;
    
    if(str in stringToPatternMap && stringToPatternMap[str] !== patternChar)
      return false;
    
    stringToPatternMap[str] = patternChar;
    patternToStringMap[patternChar] = str;
  }
  return true;    
};